go/buildhelper/client.go

```diff
@@ -0,0 +1,96 @@
+package buildhelper
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/obscuro-playground/go/ethclient"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+type BuildHelperClient struct {
+	node *EthNode
+}
+
+func (b *BuildHelperClient) FetchBlockByNumber(n *big.Int) (*types.Block, error) {
+	return b.node.FetchBlockByNumber(n)
+}
+
+func (b *BuildHelperClient) FetchBlock(id common.Hash) (*types.Block, error) {
+	return b.node.FetchBlock(id)
+}
+
+func (b *BuildHelperClient) FetchHeadBlock() (*types.Block, uint64) {
+	blk, err := b.node.FetchBlockByNumber(nil)
+	if err != nil {
+		panic(err)
+	}
+	return blk, blk.Number().Uint64()
+}
+
+func (b *BuildHelperClient) IssueTx(tx obscurocommon.EncodedL1Tx) {
+	b.node.BroadcastTx(tx)
+}
+
+func (b *BuildHelperClient) Info() ethclient.Info {
+	return ethclient.Info{ID: b.node.id}
+}
+
+func (b *BuildHelperClient) BlocksBetween(startingBlock *types.Block, lastBlock *types.Block) []*types.Block {
+	//TODO this should be a stream
+	var blocksBetween []*types.Block
+	var err error
+	for currentBlk := lastBlock; currentBlk != nil || currentBlk.Hash() != startingBlock.Hash(); {
+		if currentBlk.ParentHash() == common.HexToHash("") {
```

Comment:
@otherview Isn't this an error condition? You asked for the blocks between `startingBlock` and `lastBlock`, but the chain ended before `startingBlock` was found.

Reply:
Yaa, needed a breakpoint there 😸